File: gui/src/store/schemas/manifest.ts
export interface Manifest {
    schemaVersion: number;
    config: descriptor;
    layers: descriptor[];
    mediaType: string;
    annotations?: annotations;
}

interface descriptor {
    mediaType: string;
    digest: string;
    size: string;
    urls?: string[];
    annotations?: annotations;
}

interface annotations {
    [key: string]: string;
}

export const defaultManifestSchema: Manifest = {
    schemaVersion: 2,
    config: {
        mediaType: "",
        digest: "",
        size: "",
    },
    layers: [],
    mediaType: "",
};

File: gui-wasm/src/types/manifest.rs
use std::collections::HashMap;

use serde::{Serialize, Deserialize};
use std::default::Default;

#[derive( Serialize, Deserialize, Debug, Clone, Default)]
#[serde(default, rename_all = "camelCase")]
pub struct Manifest {
    pub schema_version: u8,
    pub config: Descriptor,
    pub layers: Vec<Descriptor>,
    pub media_type: String,
    pub annotations: HashMap<String, String>,
}

#[derive(Serialize, Deserialize, Debug, Clone, Default)]
#[serde(default, rename_all = "camelCase")]
pub struct Descriptor {
    pub media_type: String,
    pub digest: String,
    pub size: u32,
    pub urls: Vec<String>,
    pub annotations: HashMap<String, String>
}


